syft/messaging/plan.py

```diff
@@ -99,30 +171,38 @@ class Plan(ObjectStorage):
 
     def __init__(
         self,
-        id: Union[str, int],
-        owner: "sy.workers.BaseWorker",
+        id: Union[str, int] = None,
+        owner: "sy.workers.BaseWorker" = None,
         name: str = "",
+        state_ids: List[Union[str, int]] = None,
         arg_ids: List[Union[str, int]] = None,
         result_ids: List[Union[str, int]] = None,
-        blueprint: callable = None,
         readable_plan: List = None,
-        is_method: bool = False,
+        blueprint=None,
+        state=None,
+        include_state: bool = False,
         is_built: bool = False,
         verbose: bool = False,
         *args,
         **kwargs,
     ):
-        super().__init__()
+        ObjectStorage.__init__(self)
+        torch.nn.Module.__init__(self)
 
         # Plan instance info
-        self.id = id
-        self.name = name
-        self.owner = owner
+        self.id = sy.ID_PROVIDER.pop() if id is None else id
+        self.name = self.__class__.__name__ if name == "" else name
+        self.owner = sy.local_worker if owner is None else owner
         self.verbose = verbose
 
         # Info about the plan stored
         self.plan = list()
         self.readable_plan = readable_plan if readable_plan is not None else []
+        self.state = State(self)
+        self.include_state = include_state
+        if state is not None:
```

Comment:
A state can store another state?

Reply:
Not at all, but it can be fed with a dict like in self.state.set_(state)